oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/OAuth2ClientAuthorizationProvider.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.oauth2.client;
+
+/**
+ *
+ * @author Joe Grandja
+ * @since 5.2
+ */
+public interface OAuth2ClientAuthorizationProvider<T extends AbstractOAuth2ClientAuthorizationContext> {
```

Comment:
I actually did consider the name `OAuth2AuthorizedClientManager` but decided otherwise since it doesn't really **manage** `OAuth2AuthorizedClient` instances. It's sole purpose it to authorize a client and therefore **provide** a new `OAuth2AuthorizedClient`. As far as managing  `OAuth2AuthorizedClient`(s), this is the role of the `OAuth2AuthorizedClientService`.

So I don't feel adding `Manager` to the name makes sense.

I also considered the name `OAuth2AuthorizedClientProvider` but liked `OAuth2ClientAuthorizationProvider` as it aligned with `AuthenticationProvider`. However, I still think `OAuth2AuthorizedClientProvider` could work as well.

Reply:
I can by into `Provider`, but I strongly believe it should start with `OAuth2AuthorizedClient` since that is what it is providing.

For what it is worth, I believe that `OAuth2ClientAuthorizationProvider` does not align with `AuthenticationProvider`. `AuthenticationProvider` returns an `Authentication`. This API returns an `OAuth2AuthorizedClient` and not an `OAuth2ClientAuthorization` nor a `OAuth2Client` as the name `OAuth2ClientAuthorizationProvider` would imply.

Reply:
Ok I'll rename to `OAuth2AuthorizedClientProvider`

Reply:
Thanks. When you do can you rename the PR subject too please?